icmplib/ping.py

```diff
@@ -182,3 +185,79 @@ def ping(address, count=4, interval=1, timeout=2, id=PID, source=None,
     socket.close()
 
     return host
+
+
+async def aioping(address, count=4, interval=1, timeout=2, packet_id=0, source=None, privileged=True, **kwargs):
+    """
+    Send ICMP Echo Request packets to a network host.
+
+    Same API as `ping`.
+    Usage::
+        >>> import asyncio
+        >>> from icmplib import aioping
+        >>> asyncio.run(aioping('1.1.1.1'))
+        Or
+        >>> async def my_func():
+        >>>     host = await aioping('1.1.1.1')
+        >>>     host.avg_rtt
+        >>>     host.is_alive
+        >>> asyncio.run(my_func())
+        13.2
+        True
+
+    See the `Host` class for details.
+    """
+    address = resolve(address)
+
+    try:
+        IPv6Address(address)
+        sock = ICMPv6Socket(address=source, privileged=privileged)
+    except AddressValueError:
+        sock = ICMPv4Socket(address=source, privileged=privileged)
+    sock._socket.setblocking(False)  # Should probably be moved down to the Socket API?
+    packets_sent = 0
+    packets_received = 0
+
+    min_rtt = float('inf')
+    avg_rtt = 0.0
+    max_rtt = 0.0
+    packet_id = random.randint(0, 100000) if not packet_id else packet_id
+    loop = asyncio.get_event_loop()
+    for sequence in range(count):
+        request = ICMPRequest(destination=address, id=packet_id, sequence=sequence, **kwargs)
+        try:
+            await loop.run_in_executor(None, partial(sock.send, request))
+            packets_sent += 1
+            reply = await loop.run_in_executor(None, partial(sock.receive, request, timeout))
```

Comment:
Should probably also look into the approach I did for the other port, as seen [here](https://github.com/JonasKs/aioping3/blob/main/aioping3/aioping3.py#L192-L194).